processing/tests/conftest.py

```diff
@@ -67,24 +74,43 @@ def files_data_for_pipeline(files_data):
     return files_data
 
 
+alembic_cfg = Config("alembic.ini")
+
+
+@pytest.fixture(scope="session")
+def use_temp_env_var():
+    with patch.dict(os.environ, {"USE_TEST_DB": "1"}):
+        yield
+
+
 @pytest.fixture
-def setup_database():
-    test_db_name = uuid.uuid4().hex
-    db_url = f"sqlite:///./{test_db_name}.db"
-    engine = create_engine(db_url, connect_args={"check_same_thread": False})
-    Base.metadata.create_all(bind=engine)
-    session_local = sessionmaker(
-        autocommit=False, autoflush=False, bind=engine
-    )
-    session = session_local()
-    yield session
-    session.close()
-    Base.metadata.drop_all(bind=engine)
-    remove_db(test_db_name)
+def test_db_engine():
+    db_url = get_test_db_url(settings.database_url)
+    test_db_engine = create_engine(db_url)
+    yield test_db_engine
 
 
-def remove_db(db_name):
-    base = Path(__file__).parent.parent.parent
-    dbs = base.glob(f"**/{db_name}.db")
-    for db in dbs:
-        db.unlink()
+@pytest.fixture
+def setup_database(use_temp_env_var, test_db_engine):
+
+    if not database_exists(test_db_engine.url):
+        create_database(test_db_engine.url)
+
+    try:
+        command.upgrade(alembic_cfg, "head")
+    except SQLAlchemyError as e:
+        raise SQLAlchemyError(f"Got an Exception during migrations - {e}")
+
+    yield
+
+    drop_database(test_db_engine.url)
+
+
+@pytest.fixture
+def test_db_session(test_db_engine, setup_database):
```

Comment:
May you rename it the same way as testing engine?

Reply:
agreed - renamed to db_test_session